airbyte-cdk/python/airbyte_cdk/sources/file_based/config/abstract_file_based_spec.py

```diff
@@ -17,6 +17,15 @@ class AbstractFileBasedSpec(BaseModel):
     that are needed when users configure a file-based source.
     """
 
+    start_date: Optional[str] = Field(
+        title="Start Date",
+        description="UTC date and time in the format 2017-01-25T00:00:00Z. Any file modified before this date will not be replicated.",
+        examples=["2021-01-01T00:00:00Z"],
+        format="date-time",
+        pattern="^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
```

Comment:
Could you use a pattern descriptor whenever you specify a pattern? https://docs.airbyte.com/connector-development/connector-specification-reference#pattern-descriptors